30413/Paltineanu Rares - Mihai/Assignment1/main.c

```diff
@@ -0,0 +1,331 @@
+#include <stdio.h>
+#include <stdlib.h>
+#include <string.h>
+
+#define command1 "AF"
+#define command2 "AL"
+#define command3 "DF"
+#define command4 "DL"
+#define command5 "DOOM_THE_LIST"
+#define command6 "DE"
+#define command7 "PRINT_ALL"
+#define command8 "PRINT_F"
+#define command9 "PRINT_L"
+
+
+int ct = 0;
+
+typedef struct array {
+
+    int element;
+    struct array* next;
+
+} arrayT;
+
+arrayT* first,* last;
+
+char* takingTheCommand(char* array) {
+
+    int i = 0;
+    char* command = (char*)malloc(30);
+
+    while(array[i] != ' ' && array[i] != '\0') {
+
+        command[i] = array[i];
+        i++;
+
+    }
+    command[i] = '\0';
+    return command;
+}
+
+void addElement(int number, int firstOrLast) {
+
+    if(firstOrLast == 1) {  ///first
+
+        if(first == NULL) {
+
+            first = (arrayT*)malloc(sizeof(arrayT));
+            first->element = number;
+            first->next = last;
+            last = first;
+
+        } else {
+
+            arrayT* current = (arrayT*)malloc(sizeof(arrayT));
+            current->element = number;
+            current->next = first;
+            first = current;
+
+        }
+    } else { ///last
+
+        arrayT* current = (arrayT*)malloc(sizeof(arrayT));
+        last->next = current;
+        current->element = number;
+        current->next = NULL;
+        last = current;
+
+    }
+}
+
+void deleteElement(int firstOrLast) {
```

Comment:
I don't like this approach *at all*!
`deleteFirst` should be a separate function form `deleteLast`. They do different things, differently. Adding booleans as parameters to a function just so that its execution differs based on that boolean thus creating multiple execution branches is a very bad practice. The choice of branches, in more conceptual and general terms should always be made as high-up in the execution chain as possible (in this case, at the moment the command is read from the file.

Reply:
I thought that it would be a good idea :)

Reply:
Should i remake it?

Reply:
Well, if you can make the case for it being a good idea, no!
If my arguments were valid enough, then yes. It's up to you, it's not a functional mistake but rather a code inadequacy (in my humble opinion)